exercises/sum-of-multiples/example.rs

```diff
@@ -0,0 +1,17 @@
+use std::collections::BTreeSet;
+
+pub fn sum_of_multiples(limit: u32, factors: &Vec<u32>) -> u32 {
```

Comment:
Not to say that I completely understand everything that goes on here, but at least for Vectors I believe it's the combination of the facts:

* [Vector](https://doc.rust-lang.org/std/vec/struct.Vec.html) implements Deref with Target `[T]`, so that `v.deref()` gives `&[T]`.
* https://doc.rust-lang.org/book/deref-coercions.html

> If you have a type `U`, and it implements `Deref<Target=T>`, values of `&U` will automatically coerce to a `&T`

Let `U = Vec<E>` and `T = [E]`.

* the condition is met. We have a type `Vec<E>`, and it implements `Deref<Target=[E]>`
* Thus, values of `&Vec<E>` coerce to `&[E]`
* And that's what we are doing in our tests. This function takes a `&[u32]`, but we pass a `&Vec<u32>`, but it works.